types/decimal.go

```diff
@@ -551,6 +551,25 @@ func (d Dec) Ceil() Dec {
 
 //___________________________________________________________________________________
 
+// ValidSortableDec ensures that a Dec is within the sortable bounds,
+// a Dec can't have a precision of less than 10^-18.
+// Max sortable decimal was set to the reciprocal of SmallestDec.
+func ValidSortableDec(dec Dec) bool {
+	return dec.LTE(OneDec().Quo(SmallestDec()))
+}
+
+// SortableDecBytes returns a byte slice representation of a Dec that can be sorted.
+// Left and right pads with 0s so there are 18 digits to left and right of the decimal point.
+// For this reason, there is a maximum and minimum value for this, enforced by ValidSortableDec.
+func SortableDecBytes(dec Dec) []byte {
+	if !ValidSortableDec(dec) {
+		panic("dec must be within bounds")
+	}
+	return []byte(fmt.Sprintf(fmt.Sprintf("%%0%ds", Precision*2+1), dec.String()))
```

Comment:
I don't think use of `fmt.Sprintf` is the most efficient way of doing this. Why not operate on the bytes directly?

Reply:
Because the Sprintf method has an easy formatting code to left pad with the number of zeros we want.  What do you man by operate on the bytes directly?  Is Sprintf that expensive?

Reply:
Because the Sprintf method has an formatting code to really easily left pad with the number of zeros we want.  Is Sprintf really that expensive?

Reply:
I originally had the the inner Sprintf as an in-code const to avoid doing it everytime, but @jackzampolin suggested moving it into the code function.

Reply:
This padding scheme won't always produce fix length byte arrays. 
1. The largest input 1000000000000000000 has length 38, all other positive inputs have length 37.
2. Currently, negative inputs are unbounded. 

I think we should add one additional padding or change `ValidSortableDec` to strictly less than, as well as verifying the absolute value of the input to account for large negative numbers.

Reply:
Ah shit, didn't consider negative numbers (originally did this for prices).  Ok, will update.

Reply:
Good catches.  For #1, should I add an extra 0 to the front of everything, or just make max sortable < than reciprocal?

Reply:
Addressed.